source/val/validate_interfaces.cpp

```diff
@@ -533,6 +533,64 @@ spv_result_t ValidateLocations(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateStorageClass(ValidationState_t& _,
+                                  const Instruction* entry_point) {
+  bool has_push_constant = false;
+  bool has_ray_payload = false;
+  bool has_hit_attribute = false;
+  bool has_callable_data = false;
+  for (uint32_t i = 3; i < entry_point->operands().size(); ++i) {
+    auto interface_id = entry_point->GetOperandAs<uint32_t>(i);
+    auto interface_var = _.FindDef(interface_id);
+    auto storage_class = interface_var->GetOperandAs<SpvStorageClass>(2);
+    switch (storage_class) {
+      case SpvStorageClassPushConstant: {
+        if (has_push_constant) {
+          return _.diag(SPV_ERROR_INVALID_DATA, entry_point)
+                 << _.VkErrorID(6673)
+                 << "Entry-point has more than one variable with the "
+                    "PushConstant storage class in the interface";
+        }
+        has_push_constant = true;
```

Comment:
What's Vulkan's stance on duplicate information generally? VUID 6674 says you can't use more than one variable with a push constant storage class per entry point. That covers all module versions. This seems redundant, but only covering SPIR-V 1.4 or later. There is code in validate_decorations.cpp for VUID 6674. To me, this duplication is not great from a maintenance point of view.